### PR TITLE
dockerfile: change base image to rockylinux

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4287

### What is changed and how it works?
- change base image to rockylinux

#### Tests 

 - chaos test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]
LGTM

File: dm/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,4 @@
 FROM golang:1.19.1-alpine3.15 as builder
```
Comment (lance6716) [inline dm/Dockerfile:1]: do we need to update this image? not sure if the building stage should also be put in safer image

File: dm/Dockerfile [hunk 2]
```diff
@@ -18,7 +17,7 @@ RUN apk update && apk add bash
 
 RUN make dm-worker dm-master dmctl
 
-FROM alpine:3.15
+FROM rockylinux:9.0
```
Comment (CharlesCheung96) [inline dm/Dockerfile:20]: Why should this change be made? Should the relevant dockerfile in engine be changed to this as well?

File: dm/Dockerfile (hunk 1 above)
Comment (lijie0123) [reply dm/Dockerfile:1]: If runtime use glibc, builder use glibc

Comment (lijie0123) [reply dm/Dockerfile:1]: here we just parallel produce another docker image, directly change may be dangerous

**Review by lijie0123 — CHANGES_REQUESTED** [review]

**Comment by lijie0123** [comment]
here we just parallel produce another docker image, directly change may be dangerous， so you can add this dockerfile to https://pingcap.feishu.cn/wiki/wikcnYlK7ZLs709jXDFSBlUHm6f

File: dm/Dockerfile (hunk 2 above)
Comment (amyangfei) [reply dm/Dockerfile:20]: Yes, engine will replace alpine with rockylinux too.

**Pull request closed without merging**